### PR TITLE
Fix flacky integration tests

### PR DESCRIPTION
**Pull request by Shaddoll**
**What changed?**

**Why?**
The tests have race conditions.

**How did you test it?**

**Potential risks**

**Release notes**

**Documentation Changes**

### CODE REVIEW

File: host/queryworkflow_test.go [hunk 1]
```diff
@@ -613,7 +615,7 @@ func (s *IntegrationSuite) TestQueryWorkflow_Consistent_PiggybackQuery() {
 		} else if previousStartedEventID > 0 {
 			for _, event := range history.Events[previousStartedEventID:] {
 				if *event.EventType == types.EventTypeWorkflowExecutionSignaled {
-					handledSignal = true
+					handledSignal.Store(true)
```
Comment (mkolodezny) [inline host/queryworkflow_test.go:618]: nit: do we need to be sure to set it just once? then please consider sync.Once

File: host/queryworkflow_test.go [hunk 2]
```diff
@@ -961,7 +964,7 @@ func (s *IntegrationSuite) TestQueryWorkflow_Consistent_BlockedByStarted_NonStic
 				if *event.EventType == types.EventTypeWorkflowExecutionSignaled {
 					// wait for some time to force decision task to stay in started state while query is issued
 					<-time.After(5 * time.Second)
-					handledSignal = true
+					handledSignal.Store(true)
```
Comment (mkolodezny) [inline host/queryworkflow_test.go:967]: idem

File: host/queryworkflow_test.go [hunk 3]
```diff
@@ -1149,7 +1153,7 @@ func (s *IntegrationSuite) TestQueryWorkflow_Consistent_NewDecisionTask_Sticky()
...
```
Comment (mkolodezny) [inline host/queryworkflow_test.go:1156]: idem

**Review by mkolodezny — APPROVED** [review]
Thanks for fixing this!

File: host/queryworkflow_test.go (hunk 1 above)
Comment (Shaddoll) [reply host/queryworkflow_test.go:618]: I don't think it needs to be set just once.

**Pull request merged**